### ISSUE TITLE
methods(Vararg) throws

### ISSUE THREAD

**Issue by tkf** [JuliaLang/julia#29645]
```julia
julia> methods(Vararg)
ERROR: TypeError: in Type, in parameter, expected Type, got Type{Vararg}
Stacktrace:
 [1] signature_type(::Any, ::Any) at ./reflection.jl:662
 [2] _methods at ./reflection.jl:713 [inlined]
 [3] methods(::Any, ::Any) at ./reflection.jl:758
 [4] methods(::Any) at ./reflection.jl:773
 [5] top-level scope at none:0
```

It looks like the error happens at

https://github.com/JuliaLang/julia/blob/9fd9962a5d0c983dd6dc1c2491f67bb3b1bffd1f/base/reflection.jl#L661-L662

where `signature_type` tries to get `Type{Vararg}`.

Maybe check if `f <: Vararg` in `methods(f, t)` and return an empty `MethodList`?  I'm not sure if it is worth a special handling, though.  The document didn't say anything about type of `f` (but I've been assuming that `methods(::Base.Callable, [::Type])`) so this probably is not a bug at all.

**Comment by JeffBezanson** [comment]
I think the error is correct. `Vararg` by itself is not a well-formed type; it's not that it has no methods, but that it doesn't make sense to ask whether it has methods.

**Comment by tkf** [comment]
> `Vararg` by itself is not a well-formed type

Then why `Vararg isa Type` is evaluated to `true`?

**Comment by JeffBezanson** [comment]
It very likely shouldn't be, similar to `TypeVar`. We just happen to use something in the `Type` hierarchy as its representation. However there are also other cases, namely types with free variables, that have Type-style representations but aren't well formed.

Why do you need `methods(Vararg)` to not be an error?

**Comment by tkf** [comment]
I see. Thanks for the explanation.

I have a package for searching methods.  I just noticed that it tries to call `methods(Vararg)` since it is called for any object `f` if `f isa Base.Callable`.   I fixed it for `Vararg` so it's OK for me as-is: https://github.com/tkf/InteractiveCodeSearch.jl/commit/cc91fa4efe97073c61e8c5f3e95a1550ba747782.  When fixing it, I just thought it'd be nice to have a property like "if `f isa Base.Callable` then `methods(f)` does not throw" but I didn't think about the "malformed types."

Admittedly, this is a rare case that manual handling is not so bad. I'm closing the issue.
